neard/src/cli.rs

```diff
@@ -0,0 +1,335 @@
+use super::{DEFAULT_HOME, NEARD_VERSION, NEARD_VERSION_STRING, PROTOCOL_VERSION};
+use clap::{AppSettings, Clap};
+use near_primitives::types::{NumSeats, NumShards};
+use nearcore::get_store_path;
+use std::net::SocketAddr;
+use std::path::{Path, PathBuf};
+use std::{env, fs, io};
+#[cfg(feature = "adversarial")]
+use tracing::error;
+use tracing::info;
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::EnvFilter;
+
+/// NEAR Protocol Node
+#[derive(Clap)]
+#[clap(version = NEARD_VERSION_STRING.as_str())]
+#[clap(setting = AppSettings::SubcommandRequiredElseHelp)]
+pub(super) struct NeardCmd {
+    #[clap(flatten)]
+    opts: NeardOpts,
+    #[clap(subcommand)]
+    subcmd: NeardSubCommand,
+}
+
+impl NeardCmd {
+    pub(super) fn parse_and_run() {
+        let neard_cmd = Self::parse();
+        neard_cmd.opts.init();
+        info!(target: "near", "Version: {}, Build: {}, Latest Protocol: {}", NEARD_VERSION.version, NEARD_VERSION.build, PROTOCOL_VERSION);
+
+        #[cfg(feature = "adversarial")]
+        {
+            error!("THIS IS A NODE COMPILED WITH ADVERSARIAL BEHAVIORS. DO NOT USE IN PRODUCTION.");
+
+            if env::var("ADVERSARY_CONSENT").unwrap_or_else(|_| "".to_string()) != "1" {
+                error!("To run a node with adversarial behavior enabled give your consent by setting variable:");
+                error!("ADVERSARY_CONSENT=1");
+                std::process::exit(1);
+            }
+        }
+
+        let home_dir = neard_cmd.opts.home;
+
+        match neard_cmd.subcmd {
+            NeardSubCommand::Init(cmd) => cmd.run(&home_dir),
+            NeardSubCommand::Testnet(cmd) => cmd.run(&home_dir),
+            NeardSubCommand::Run(cmd) => cmd.run(&home_dir),
+
+            NeardSubCommand::UnsafeResetData => {
+                let store_path = get_store_path(&home_dir);
+                info!(target: "near", "Removing all data from {}", store_path);
+                fs::remove_dir_all(store_path).expect("Removing data failed");
```

Comment:
Wow, couldn't it be dangerous? Shall we validate `store_path` to be meaningful file?

Reply:
Yeah, it is great that now we can easily spot such cruel shortcuts. Let's create a separate issue to address this since the implementation is not new, and it has been part of neard for quite a long time already. I guess we should printing the folder that is about to be removed and add a confirmation dialog.

Reply:
#4369